src/charm.py

```diff
@@ -93,21 +125,22 @@ def _configure_smtp_auth(
         self.unit.open_port("tcp", 465)
         self.unit.open_port("tcp", 587)
 
+        if not systemd.service_running("dovecot"):
+            systemd.service_resume("dovecot")
+            return
+
         if changed:
             self.unit.status = ops.MaintenanceStatus("Restarting Dovecot due to config changes")
             systemd.service_reload("dovecot")
```

Comment:
I believe reload shouldn't interrupt the service, so I suspect you can reload always, regardless of whether the file has changed